sky/services/semantics/semantics.mojom

```diff
@@ -5,26 +5,33 @@
 [DartPackage="sky_services"]
 module semantics;
 
+enum SemanticAction {
+  TAP,
+  LONG_PRESS,
+  SCROLL_LEFT,
+  SCROLL_RIGHT,
+  SCROLL_UP,
+  SCROLL_DOWN,
+  INCREASE,
+  DECREASE,
+};
+
 struct SemanticsNode {
   uint32 id;
 
   // A null field means "unchanged since last time".
   SemanticFlags? flags;
   SemanticStrings? strings;
   SemanticGeometry? geometry;
+  array<SemanticAction>? actions;
   array<SemanticsNode>? children;
 };
 
 struct SemanticFlags {
   // This is intended to just be booleans, so that it can be extended
   // over time yet still be packed tightly.
-  bool canBeTapped = false;
-  bool canBeLongPressed = false;
-  bool canBeScrolledHorizontally = false;
-  bool canBeScrolledVertically = false;
-  bool hasCheckedState = false; // whether isChecked is relevant
-  bool isChecked = false;
-  bool isAdjustable = false;
+  bool has_checked_state = false; // whether is_checked is relevant
+  bool is_checked = false;
```

Comment:
moving these to an array of enums certainly makes this look saner.

For the record, the original design was motivated by the way that the data would be bitpacked at the mojo layer. I don't think that'll happen for an array of enums.

That was almost certainly a premature optimisation on my part though, like the nullability here.